src/Compilers/Core/Portable/SourceGeneration/GeneratorDriver.cs

```diff
@@ -155,10 +156,12 @@ static ImmutableArray<GeneratedSourceResult> getGeneratorSources(GeneratorState
 
         internal GeneratorDriverState RunGeneratorsCore(Compilation compilation, DiagnosticBag? diagnosticsBag, CancellationToken cancellationToken = default)
         {
+            using var timer = CodeAnalysisEventSource.Log.CreateGeneratorDriverRunTimer();
```

Comment:
>using var timer [](http://example.com/codeflow?start=12&length=15)

Consider moving after the `if (_state.Generators.IsEmpty) { ... }` case below, or use `timer.Elapsed` in that return value.